webpack/scenes/ContentViews/Create/CreateContentViewForm.js

```diff
@@ -0,0 +1,165 @@
+
+const CreateContentViewForm = ({ setModalOpen }) => {
+  const dispatch = useDispatch();
+  const [name, setName] = useState('');
+  const [label, setLabel] = useState('');
+  const [description, setDescription] = useState('');
+  const [composite, setComposite] = useState(false);
+  const [component, setComponent] = useState(true);
+  const [autoPublish, setAutoPublish] = useState(false);
+  const [importOnly, setImportOnly] = useState(false);
+  const [dependencies, setDependencies] = useState(false);
+  const [redirect, setRedirect] = useState(false);
+  const [saving, setSaving] = useState(false);
+
+  const response = useSelector(selectCreateContentViews);
+  const status = useSelector(selectCreateContentViewStatus);
+  const error = useSelector(selectCreateContentViewError);
+
+  useEffect(() => {
+    const { id } = response;
+    if (id && status === STATUS.RESOLVED) {
+      setSaving(false);
+      setRedirect(true);
+    } else if (status === STATUS.ERROR) {
+      setSaving(false);
+    }
+  }, [JSON.stringify(response), status, error]);
+
+  const onSave = () => {
+    setSaving(true);
+    dispatch(createContentView({
+      name,
+      label,
+      description,
+      composite,
+      dependencies,
+      auto_publish: (autoPublish && composite),
+      import_only: importOnly,
+    }));
+  };
+
+  useEffect(
+    () => {
+      setLabel(name.replace(/ /g, '_'));
+    },
+    [name],
+  );
+
+  if (redirect) {
+    const { id } = response;
+    return (<Redirect to={`/labs/content_views/${id}`} />);
+  }
+
+  return (
+    <Form>
+      <FormGroup label="Name" isRequired fieldId="name">
+        <TextInput
+          isRequired
+          type="text"
+          id="name"
+          aria-label="input_name"
+          name="name"
+          value={name}
+          onChange={value => setName(value)}
+        />
+      </FormGroup>
+      <FormGroup label="Label" isRequired fieldId="label">
+        <TextInput
+          isRequired
+          type="text"
+          id="label"
+          aria-label="input_label"
+          name="label"
+          value={label}
+          onChange={value => setLabel(value)}
+        />
+      </FormGroup>
+      <FormGroup label="Description" fieldId="description">
+        <TextArea
+          isRequired
+          type="text"
+          id="description"
+          name="description"
+          aria-label="input_description"
+          value={description}
+          onChange={value => setDescription(value)}
+        />
+      </FormGroup>
+      <FormGroup isInline fieldId="component">
+        <Radio
+          id="component"
+          name="component"
+          role="radio"
+          label={LabelComponent()}
+          isChecked={component}
+          onChange={(checked) => { setComponent(checked); setComposite(!checked); }}
+          description="Consists of repositories"
```

Comment:
not sure if I missed this, but I really like the simplicity of these descriptions! :+1: